vac/raw/eth-mls-offchain.md

```diff
@@ -0,0 +1,225 @@
+
+Since the de-MLS consists of MLS backend, the MLS services and components
+often used in following specification as it is or modified.
+
+### MLS Services
+
+MLS is operated in two services authentication service (AS) and delivery service (DS).
+AS enables group members to authenticate the credentials presented by other group members
+while DS routes MLS messages among the nodes or members in the protocol in the correct
+order and manage the `keyPackage` of the users where the `keyPackage` is the objects
+that provide some public information about a user.
+
+### MLS Objects
+
+Following section presents the MLS objects and components that used in this RFC:
+
+`Epoch`: Fixed time intervals that changes the state that is defined by members,
+section 3.4 in [MLS RFC 9420](https://datatracker.ietf.org/doc/rfc9420/).
+
+`MLS proposal message:` Members MUST receive the proposal message prior to the
+corresponding commit message that initiates a new epoch with key changes,
+in order to ensure the intended security properties, section 12.1 in [MLS RFC 9420](https://datatracker.ietf.org/doc/rfc9420/).
+In this RFC, the add and remove proposals are used.
+
+`Application message`: This message type used in arbitrary encrypted communication between group members.
+This is restricted by [MLS RFC 9420](https://datatracker.ietf.org/doc/rfc9420/) as if there is pending proposal,
+the application message should be cut.
+Notet that: Since the MLS is based on servers, this delay between proposal and commit messages are very small.
+`Commit message:` After members receive the proposals regarding group changes,
+the committer, who may be any member of the group, as specified in  [MLS RFC 9420](https://datatracker.ietf.org/doc/rfc9420/),
+generates the necessary key material for the next epoch, including the appropriate welcome messages
+for new joiners and new entropy for removed members. In this RFC, the committers only MUST be stewards.
+
+### de-MLS Objects
+
+`Voting proposal message:` Voting proposal messages are different then MLS proposals as voting proposals are basically
+the application message in MLS group. Therefore, the steward can collect the proposals without halting the protocol.
+
+## Flow
+
+General flow is as follows:
+
+- Each time a single `steward` initializes a group and creates and emits a group anouncement (GA) periodically.
+- Meanwhile, each`node`creates and sends their`credential` includes `keyPackage`.
+- Each `member`creates `voting proposals` sends them to from MLS group during epoch E.
+- Meanwhile, the `steward` collects finalized `voting proposals` from MLS group and converts them into
+`MLS proposals`  then sends them with correspondng `commit messages`
+- Evantually, with the commit messages, all members starts the next epoch E+1.
+
+## Creating Voting Proposal
+
+A `member` MAY initializes the voting with the proposal payload
+which is implemented using [protocol buffers v3](https://protobuf.dev/) as follows:

```

Comment:
Look here I disagree, in fact only stewards start voting, then they send voting proposal and then everyone votes. 


Are we sure we want to add to the rfc that anyone can start voting?